### DISCUSSION TITLE
--db-dir size limit?

### DISCUSSION THREAD

**Discussion by acuozzo** [readysettech/readyset#43] [Q&A] [answered]
The systems I intend to run ReadySet on have a much smaller amount of space available to them than my DB server does.

With --durability=persistent, does ReadySet just keep going until it exhausts --db-dir?

Thanks!

**Comment by glittershark** [comment, marked as answer]
ReadySet needs to fully snapshot the tables in your upstream database in order to be able to cache query results that target those tables - this will take at *least* the amount of space as your db server, but likely more.

You can limit the specific tables ReadySet snapshots by passing a list of tables to the `--replication-tables` command line flag when running ReadySet, though then we'll only be able to cache queries that mention those tables

**Reply by acuozzo** [reply]
Got it! Thank you for replying.

ReadySet won't fit my use-case then, unfortunately. Damn! :-(

I may wind up writing a VFS for SQLite instead which keeps a limited cache of pages and swaps in from my DB on demand.

My ultimate goal is to keep the hottest e.g. 10 GiB of my database local.
